terraform/table_terraform_module.go

```diff
@@ -0,0 +1,135 @@
+package terraform
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"reflect"
+
+	"github.com/Checkmarx/kics/pkg/model"
+	"github.com/turbot/steampipe-plugin-sdk/v5/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
+)
+
+func tableTerraformModule(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "terraform_module",
+		Description: "Terraform module information.",
+		List: &plugin.ListConfig{
+			ParentHydrate: tfConfigList,
+			Hydrate:       listModules,
+			KeyColumns:    plugin.OptionalColumns([]string{"path"}),
+		},
+		Columns: []*plugin.Column{
+			{
+				Name:        "name",
+				Description: "Module name.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "source",
+				Description: "Module source",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "version",
+				Description: "Module version",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "start_line",
+				Description: "Starting line number.",
+				Type:        proto.ColumnType_INT,
+			},
+			{
+				Name:        "path",
+				Description: "Path to the file.",
+				Type:        proto.ColumnType_STRING,
+			},
+		},
+	}
+}
+
+type terraformModule struct {
+	Name      string
+	Path      string
+	StartLine int
+	Source    string
+	Version   string
+}
+
+func listModules(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	// The path comes from a parent hydrate, defaulting to the config paths
+	// or available by the optional key column
+	path := h.Item.(filePath).Path
+
+	combinedParser, err := Parser()
+	if err != nil {
+		plugin.Logger(ctx).Error("terraform_module.listModules", "create_parser_error", err)
+		return nil, err
+	}
+
+	content, err := os.ReadFile(path)
+	if err != nil {
+		plugin.Logger(ctx).Error("terraform_module.listModules", "read_file_error", err, "path", path)
+		return nil, err
+	}
+
+	var tfModule terraformModule
+
+	for _, parser := range combinedParser {
+		parsedDocs, err := ParseContent(ctx, d, path, content, parser)
```

Comment:
Does the parser pick up any of the other arguments in a `module` block? For instance, from https://developer.hashicorp.com/terraform/language/modules/syntax#calling-a-child-module, they seem to support input variables, e.g., `servers` in this example:
```
module "consul" {
  source  = "hashicorp/consul/aws"
  version = "0.0.5"

  servers = 3
}
```

There are also a few other meta-arguments each `module` can contain, e.g., `count`, `for_each`, `providers`, `depends_on` (from https://developer.hashicorp.com/terraform/language/modules/syntax#meta-arguments).

The input variables and meta-arguments would be valuable to add IMO, as the current set of columns feel a bit light based on the possible arguments.

Reply:
Thanks much for the feedback.  The meta-arguments I added; basically just re-using what was done in `table_terraform_data_source.go` for meta-arguments.

As for the variables, not too sure to be honest i.e. I am struggling a bit to think of a scenario/use-case where and how I would use them if they were available as a column. As they are dynamic and vary per used module, it'd would probably have to be implemented similar to e.g.  https://github.com/turbot/steampipe-plugin-terraform/blob/main/terraform/table_terraform_data_source.go#L97
I can look into it though.

Reply:
@rollwagen I'm not sure how Terraform refers to them, but `arguments` seems like a good column to wrap them in (similar to `terraform_data_source`).

Does the TF parser pull these arguments in? If so, that should make it a lot easier for us to add them into a new column; however, if the parser doesn't, then we can think of an alternative approach.

Reply:
Thanks for the input @cbruno10  The TF parser does indeed pick up the module input variables. I've added this now (see latest commit).